Loader/ComboLoader.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * File containing the ComboLoader class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\Loader;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use Symfony\Component\Templating\EngineInterface;
+
+/**
+ * ComboLoader combines JS or CSS files.
+ */
+class ComboLoader implements Loader
+{
+    /**
+     * @var string Path to YUI build directory.
+     */
+    private $yuiBuildDirectory;
+
+    /**
+     * @var string Path to web directory.
+     */
+    private $webDirectory;
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \Symfony\Component\Templating\EngineInterface */
+    private $templating;
+
+    /**
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     * @param \Symfony\Component\Templating\EngineInterface $templating
+     * @param string $yuiBuildDirectory
+     * @param string $webDirectory
+     */
+    public function __construct(ConfigResolverInterface $configResolver, EngineInterface $templating, $yuiBuildDirectory, $webDirectory)
+    {
+        $this->configResolver = $configResolver;
+        $this->templating = $templating;
+        $this->yuiBuildDirectory = $yuiBuildDirectory;
+        $this->webDirectory = $webDirectory;
+    }
+
+    public function combineFilesContent(array $files)
+    {
+        $content = '';
+
+        foreach ($files as $file) {
+            $ext = $this->findFileExtension($file);
+
+            if ($ext === Loader::EXT_JS) {
+                if ($this->isTemplateFile($file)) {
```

Comment:
The `if` is getting a bit too big imgo.

I'd abstract the type test (`$ext === Loader::EXT_JS`) into meaningful methods:
```php
if ($this->isTemplateFile($file)) {
    $content .= $this->renderTemplate($file);
} elseif ($this->isYuiFile($file) || $this->isCssFile($file)) {
    $file = $this->fixUnresolvedPath($file);
    $content .= $this->loadFile($file, $ext);
} else {
    throw new NotFoundException('file type', $file);
}
```